viewing_party/party.py

```diff
@@ -0,0 +1,109 @@
+def create_movie(title, genre, rating):
+    if not (title and genre and rating):
+        return None
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating,
+    }
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+        
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watched_list = user_data["watched"]
+    to_watch_list = user_data["watchlist"]
+    for movie_dict in to_watch_list:
+        if movie_dict["title"] == title:
+            to_watch_list.remove(movie_dict)
+            watched_list.append(movie_dict)
+    return user_data
+  
+def get_watched_avg_rating(user_data):
+    users_watched_list = user_data["watched"]
+    ratings_list = []
+    if len(users_watched_list) == 0:
+        return 0.0
+    for movie in users_watched_list:
+        movie_rating = movie["rating"]
+        ratings_list.append(movie_rating)
+        average_rating = sum(ratings_list)/len(ratings_list)
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    watched_movie_data = user_data["watched"]
+    genre_list = []
+    if len(watched_movie_data) == 0:
+        return None
+    for movie in watched_movie_data:
+        genre_of_movie = movie["genre"]
+        genre_list.append(genre_of_movie)
+    genre_count = {}
+    for genre in genre_list:
+        genre_count[genre] = genre_list.count(genre)
+    most_watched_genre = max(genre_count, key=genre_count.get)
+    return most_watched_genre
```

Comment:
Great use of resources to find other methods we don't use in our curriculum. In the future, just add a comment that describes that code that we don't cover in the curriculum showing you understand its use.